api/data/reviews.js

```diff
@@ -0,0 +1,206 @@
+        if (comment && !validate.validateString(comment))
+            throw "If you supply a comment, its gotta be a non empty string";
+
+        let word = "";
+        if (comment)
+            word = comment;
+
+        // TODO: validate date
+
+        let user;
+        let game;
+
+        try {
+            user = await userMethods.getUserById(userId);
+            game = await gameMethods.getGameById(gameId);
+        } catch (e) {
+            throw e;
+        }
+
+        const newReview = {
+            _id: ObjectId(),
+            userId: userId,
+            gameId: gameId,
+            rating: rating,
+            comment: word,
+            displayName: user.displayName,
+            datePosted: date
+        };
+
+        const reviewCollection = await reviews();
+        const insertInfo = await reviewCollection.insertOne(newReview);
+        if (insertInfo.insertedCount === 0)
+            throw "Could not create new review object";
+
+        // need to update user and game review id lists
+        let userReviews = user.reviews;
+        let gameReviews = game.reviews;
+        userReviews.push(newReview._id);
+        gameReviews.push(newReview._id);
+
+        const newUser = {
+            _id: user._id,
+            firstName: user.firstName,
+            lastName: user.lastName,
+            displayName: user.displayName,
+            email: user.email,
+            favoriteGames: user.favoriteGames,
+            reviews: userReviews,
+            profilePic: user.profilePic
+        };
+
+        const userCollection = await users();
+        const gameCollection = await games();
+
+        const userUpdate = await userCollection.updateOne({ _id: userId }, { $set: newUser });
+        if (userUpdate.modifiedCount === 0)
+            throw "Could not modify user with new review";
+
+        const newGame = {
+            reviews: gameReviews,
+            endpointId: game.endpointId
+        };
+
+        const gameUpdate = await gameCollection.updateOne({ _id: gameId }, { $set: newGame });
+        if (gameUpdate.modifiedCount === 0)
+            throw "Could not modify game with new review";
+
+
+        const finalReview = await this.getReviewById(newReview._id);
+        return finalReview;
+    },
+
+    // updates the rating of the review only
+    // date is an optional parameter
+    async updateRating(reviewId, newRating, date = undefined) {
+        if (arguments.length != 2 && arguments.length != 3)
+            throw "Usage: Review Id, New Rating, OPTIONAL PARAMTER: Date";
+
+        // retrive review object
+        let review;
+        try {
+            review = await this.getReviewById(reviewId);
+        } catch (e) { throw e; }
+
+        if (!Number.isInteger(newRating) || newRating < 1 || newRating > 10)
+            throw "Rating needs to be a positive integer from 1-10";
+
+        let newDate = review.datePosted;
+        if (date)
+            newDate = date;
+
+        const changedReview = {
```

Comment:
Same as above, you can just set the single field to update based on review ID: https://docs.mongodb.com/manual/reference/operator/update/set/